### PR TITLE
C build 3

### PR DESCRIPTION
**Pull request by DenverM80**
Current build errors: 378

### CODE REVIEW

**Comment by DenverM80** [comment]
updated unit tests

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java [hunk 1]
```diff
@@ -116,18 +121,33 @@ public void generateStaticFiles() throws IOException {
...
         }
         return allStructsBuilder.build();
     }
 
+    /**
+     * Find all types that are used as an array member
+     */
+    public static ImmutableSet<String> getArrayMemberTypes(final Ds3ApiSpec spec) {
+        final ImmutableSet.Builder<String> allArrayMemberTypesBuilder = ImmutableSet.builder();
+            for (final Ds3Type ds3TypeEntry : spec.getTypes().values()) {
```
Comment (rpmoore) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java]: If you modify this slightly you wouldn't have to use a builder:

```java
return spec.getTypes().values().stream().flatMap(entry -> entry.getElements().stream).filter(element -> element.getType().equalsIgnoreCase("array")).map(element -> StructHelper.getResponseTypeName(element.getComponentType())).collect(GuavaCollectors.immutableList());

```

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/HeaderConverter.java [hunk 2]
```diff
@@ -23,10 +23,14 @@
 import com.spectralogic.ds3autogen.c.models.Header;
 import com.spectralogic.ds3autogen.c.models.Request;
 import com.spectralogic.ds3autogen.c.models.Struct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.text.ParseException;
 
 public class HeaderConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(HeaderConverter.class);
```
Comment (rpmoore) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/HeaderConverter.java]: Is this used in this file?

File: ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGenerator_Test.java [hunk 3]
```diff
@@ -152,14 +154,20 @@ public void testGenerateComplexTypedefStruct() throws IOException, ParserExcepti
         final TestFileUtilsImpl fileUtils = new TestFileUtilsImpl();
         final Map<String,Object> testMap = new HashMap<>();
 
+    /*
```
Comment (rpmoore) [inline ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGenerator_Test.java]: Why is this commented out?

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/HeaderConverter.java (hunk 2 above)
Comment (DenverM80) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/HeaderConverter.java]: It was used during development, but no longer... I'd still hesitate to remove it

**Comment by RachelTucker** [comment]
Looks good

**Pull request merged**